homeassistant/components/keba/sensor.py

```diff
@@ -0,0 +1,117 @@
+"""Support for KEBA charging station sensors."""
+import logging
+
+from homeassistant.const import ENERGY_KILO_WATT_HOUR
+from homeassistant.helpers.entity import Entity
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+SENSORS = {
+    'Curr user': ['Max current', 'mdi:flash', 'A'],
+    'Setenergy': ['Energy target', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+    'P': ['Charging power', 'mdi:flash', 'kW'],
+    'E pres': ['Session energy', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+    'E total': ['Total Energy', 'mdi:gauge', ENERGY_KILO_WATT_HOUR],
+}
+
+
+async def async_setup_platform(hass, config,
+                               async_add_entities, discovery_info=None):
+    """Set up the KEBA charging station platform."""
+    _LOGGER.debug("Initializing KEBA charging station sensors")
+    sensors = []
+    keba = hass.data[DOMAIN]
+
+    for key, _ in sorted(SENSORS.items()):
+        sensors.append(KebaSensor(key, keba))
```

Comment:
Why not pass the values to the constructor, to avoid reading `SENSORS` again down below?

Reply:
Changed according to your suggestion.